packages/block-editor/src/store/test/reducer.js

```diff
@@ -1696,60 +1696,6 @@ describe( 'state', () => {
 		} );
 
 		describe( 'blocks', () => {
-			it( 'should not reset any blocks that are not in the post', () => {
```

Comment:
The presence of the bug above was making this test pass but in reality, this behavior is something we removed a long time ago (we used to keep reusable blocks here even if they're not on the post)